code/modules/mob/mob.dm

```diff
@@ -105,6 +105,25 @@
 	else
 		to_chat(src, msg)
 
+/mob/proc/display_output(sound/S, image/I, text, turf/turf_source, vol as num, vary, frequency, falloff, channel = 0, pressure_affected = TRUE)
+	//Process sound
+	if(S && can_hear())
```

Comment:
you check can_hear() both here and in text proc
make a single can_hear check and put both cases of text and sound under it